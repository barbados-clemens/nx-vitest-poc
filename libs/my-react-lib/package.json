{
  "name": "nx-vitest-sample-lib",
  "version": "0.0.1",
  "scripts": {
    "test": "vitest run -r ."
  }
}
