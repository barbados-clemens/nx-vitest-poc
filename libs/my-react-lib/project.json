{
  "sourceRoot": "libs/my-react-lib/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/my-react-lib",
        "tsConfig": "libs/my-react-lib/tsconfig.lib.json",
        "project": "libs/my-react-lib/package.json",
        "entryFile": "libs/my-react-lib/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/my-react-lib/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/my-react-lib/**/*.{ts,tsx,js,jsx}"]
      }
    }
  }
}
