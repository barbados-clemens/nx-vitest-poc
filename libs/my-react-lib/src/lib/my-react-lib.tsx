// importing other comps seem to break stuff ðŸ¤”
import {MyReactLibTwo} from "@nx-vitest-package-scripts/my-react-lib-two";
import styles from './my-react-lib.module.css';

/* eslint-disable-next-line */
export interface MyReactLibProps {
  name?: string;
}

export function MyReactLib(props: MyReactLibProps) {
  return (
    <div className={styles['container']}>
      <h1>Welcome to MyReactLib!</h1>
      <MyReactLibTwo/>
    </div>
  );
}

export default MyReactLib;
