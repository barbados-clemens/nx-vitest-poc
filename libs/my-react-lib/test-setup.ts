// import { installGlobals } from "@remix-run/node/globals";
// import "@testing-library/jest-dom/extend-expect";
//
// installGlobals();
console.log("I am the global setup!")
