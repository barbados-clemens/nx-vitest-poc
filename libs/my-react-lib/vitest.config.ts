import {defineConfig} from "vitest/config";
import react from "@vitejs/plugin-react";
import {resolve} from "path";

export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      // TODO(caleb): how to get vite-tsconfig-paths to resolve relative to this file instead of the project root?
      "@nx-vitest-package-scripts/my-react-lib-two": resolve(__dirname, '../my-react-lib-two/src'),
    }
  },
  test: {
    include: ['**/*.test.tsx'],
    environment: "happy-dom",
    setupFiles: ["./test-setup.ts"],
  },
});
