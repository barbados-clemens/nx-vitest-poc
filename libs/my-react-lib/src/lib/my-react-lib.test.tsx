import {it, describe, expect} from 'vitest'
import renderer from 'react-test-renderer';
import {MyReactLib} from './my-react-lib'

describe("MyReactLib", () => {
  it('should work', () => {
    expect(true).toBe(true);
  })

  it('should render', () => {
    const cmp = renderer.create(<MyReactLib/>);
    let tree = cmp.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
