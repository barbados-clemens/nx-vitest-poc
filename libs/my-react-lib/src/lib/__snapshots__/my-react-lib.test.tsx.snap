// Vitest Snapshot v1

exports[`MyReactLib > should render 1`] = `
<div>
  <h1>
    Welcome to MyReactLib!
  </h1>
  <div>
    <h1>
      Welcome to MyReactLibTwo!
    </h1>
  </div>
</div>
`;
